### ISSUE TITLE
Fix nokey parameter

### ISSUE THREAD

**Issue by bkuhlmann** [troessner/reek#1666]
## Overview

I'd like to be able to define a `**nil` method parameter but this causes a stack dump.

## Steps to Recreate

You only need Reek to analyze the following code:

``` ruby
def demo(one, **nil) = puts "One: #{one}"
```

If you save the above as `snippet.rb` and run as `reek snippet.rb` you'll get the following:

<details>
  <summary>Stack Dump</summary>
  
``` 
Inspecting 1 file(s):
        !!!
        Source /Users/bkuhlmann/Engineering/OSS/marameters/test.rb cannot be processed by Reek.

        This is most likely a Reek bug.

        It would be great if you could report this back to the Reek team by opening a
        corresponding issue at https://github.com/troessner/reek/issues.

        Please make sure to include the source in question, the Reek version,
        and this entire error message, including the original exception below.

        Exception message:

        #<NoMethodError: undefined method `components' for s(:kwnilarg):#<Class:0x000000010917c898>

          children.flat_map(&:components)
                  ^^^^^^^^^>

        Original exception:

        /Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/ast/sexp_extensions/nested_assignables.rb:10:in `each'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/ast/sexp_extensions/nested_assignables.rb:10:in `flat_map'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/ast/sexp_extensions/nested_assignables.rb:10:in `components'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/ast/sexp_extensions/methods.rb:17:in `parameters'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/context/method_context.rb:53:in `default_assignments'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/smell_detectors/boolean_parameter.rb:25:in `sniff'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/smell_detectors/base_detector.rb:45:in `run'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/detector_repository.rb:44:in `block in examine'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/detector_repository.rb:42:in `each'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/detector_repository.rb:42:in `flat_map'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/detector_repository.rb:42:in `examine'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/examiner.rb:121:in `block in examine_tree'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/context/code_context.rb:56:in `each'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/context/code_context.rb:58:in `block in each'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/context/code_context.rb:57:in `each'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/context/code_context.rb:57:in `each'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/examiner.rb:120:in `flat_map'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/examiner.rb:120:in `examine_tree'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/examiner.rb:94:in `block in run'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/examiner.rb:104:in `wrap_exceptions'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/examiner.rb:93:in `run'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/examiner.rb:61:in `smells'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/examiner.rb:77:in `smelly?'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/report/progress_formatter.rb:43:in `progress'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/report/text_report.rb:22:in `add_examiner'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/cli/command/report_command.rb:26:in `block in populate_reporter_with_smells'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/cli/command/report_command.rb:25:in `each'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/cli/command/report_command.rb:25:in `populate_reporter_with_smells'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/cli/command/report_command.rb:17:in `execute'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/lib/reek/cli/application.rb:32:in `execute'
	/Users/bkuhlmann/.cache/frum/versions/3.1.2/lib/ruby/gems/3.1.0/gems/reek-6.1.1/bin/reek:12:in `<top (required)>'
	/var/folders/9s/b1_9vxs16lv8bg0dqgn15g_h0000gn/T/frum_9997_1662131172229/bin/reek:25:in `load'
	/var/folders/9s/b1_9vxs16lv8bg0dqgn15g_h0000gn/T/frum_9997_1662131172229/bin/reek:25:in `<main>'

        !!!
```
</details>

You can verify the above method signature by running the following:

``` ruby
puts method(:demo).parameters.inspect  # [[:req, :one], [:nokey]]
```

`:nokey` is the `**nil` kind.

## Environment

- **Ruby**: ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin21.4.0]
- **Reek**: 6.1.1

**Comment by mvz** [comment]
Thanks @bkuhlmann we'll look into it.

**Comment by bkuhlmann** [comment]
Thanks for the fix, looking forward to making use of this!

**Comment by bkuhlmann** [comment]
:wave: Hey, I wanted to check back on this. Would it be possible to release a new patch version with this fix or maybe you are planning to do this soon?

**Comment by troessner** [comment]
Hi Brooke,

I just released 6.1.2 with the fix in it!

**Comment by bkuhlmann** [comment]
Thanks, yep, works like a charm. :tada:
